RabbitMQ: Adding the `queue_delivery_metrics` OpenMetrics endpoint

edgrz opened the pull request:
### What does this PR do?
As part of `/detailed` metric endpoints, there's a queue delivery metrics one ([docs](https://www.rabbitmq.com/docs/prometheus#queue-delivery-metrics)), that were not integrated yet into datadog agent. This PR integrates them.

### Motivation
I prepared the DD agent to scrape such `/detailed?family=queue_delivery_metrics` but none of them were appearing in DD. So, looking into the source code I realize there was yet no mapping for there.

### Review checklist (to be filled by reviewers)

- [ ] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [ ] Add the `qa/skip-qa` label if the PR doesn't need to be tested during QA.

dkirov-dd commented:
Hello @edgrz, thank you for your contribution!
Could you add these metrics to the `metadata.csv`?

It seems like they were added in this commit: https://github.com/rabbitmq/rabbitmq-server/commit/bb93e718c2b424bdb200aa67c96b37fa2bd19600
Out of curiosity, which version of RabbitMQ are you using? Are you sure it exposes them?

Lastly, it would be great to include the new metrics in the tests as well!

edgrz commented:
Hi @dkirov-dd,

Thanks for the review. 

I'm using using rabbitMQ v4, which already exposes those. It was already part of v4.0.0, [see](https://github.com/rabbitmq/rabbitmq-server/blob/v4.0.0/deps/rabbitmq_prometheus/src/collectors/prometheus_rabbitmq_core_metrics_collector.erl#L224).

I updated manually the csv, but I'm facing an issue when setting up the dev environment locally following [docs](https://datadoghq.dev/integrations-core/setup/). By accident, when running the command `ddev config set repos.core /path/to/integrations-core` I set a wrong path and in order to fix I just run `ddev config edit` and removed the `repos.core` section. Since that moment, I always receive the following error:

```
ddev config set repos.core                                                                                                                                                              ✔  15:10:05
╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ /Users/edgar/Library/Application                                                                 │
│ Support/pyapp/ddev/10394356751862803079/11.2.0/lib/python3.12/site-packages/ddev/cli/__init__.py │
│ :165 in main                                                                                     │
│                                                                                                  │
│   162 │   manager.hook.register_commands()                                                       │
│   163 │                                                                                          │
│   164 │   try:                                                                                   │
│ ❱ 165 │   │   return ddev(prog_name='ddev', windows_expand_args=False)                           │
│   166 │   except Exception:                                                                      │
│   167 │   │   from rich.console import Console                                                   │
│   168                                                                                            │
│                                                                                                  │
│ /Users/edgar/Library/Application                                                                 │
│ Support/pyapp/ddev/10394356751862803079/11.2.0/lib/python3.12/site-packages/click/core.py:1161   │
│ in __call__                                                                                      │
│                                                                                                  │
│ /Users/edgar/Library/Application                                                                 │
│ Support/pyapp/ddev/10394356751862803079/11.2.0/lib/python3.12/site-packages/click/core.py:1082   │
│ in main                                                                                          │
│                                                                                                  │
│ /Users/edgar/Library/Application                                                                 │
│ Support/pyapp/ddev/10394356751862803079/11.2.0/lib/python3.12/site-packages/click/core.py:1694   │
│ in invoke                                                                                        │
│                                                                                                  │
│ /Users/edgar/Library/Application                                                                 │
│ Support/pyapp/ddev/10394356751862803079/11.2.0/lib/python3.12/site-packages/click/core.py:1443   │
│ in invoke                                                                                        │
│                                                                                                  │
│ /Users/edgar/Library/Application                                                                 │
│ Support/pyapp/ddev/10394356751862803079/11.2.0/lib/python3.12/site-packages/click/core.py:788 in │
│ invoke                                                                                           │
│                                                                                                  │
│ /Users/edgar/Library/Application                                                                 │
│ Support/pyapp/ddev/10394356751862803079/11.2.0/lib/python3.12/site-packages/click/decorators.py: │
│ 33 in new_func                                                                                   │
│                                                                                                  │
│ /Users/edgar/Library/Application                                                                 │
│ Support/pyapp/ddev/10394356751862803079/11.2.0/lib/python3.12/site-packages/ddev/cli/__init__.py │
│ :133 in ddev                                                                                     │
│                                                                                                  │
│   130 │   │   │   app.display_warning(error)                                                     │
│   131 │                                                                                          │
│   132 │   # Do this last                                                                         │
│ ❱ 133 │   app.set_repo(core, extras, marketplace, agent, here)                                   │
│   134 │                                                                                          │
│   135 │   # TODO: remove this when the old CLI is gone                                           │
│   136 │   app.initialize_old_cli()                                                               │
│                                                                                                  │
│ /Users/edgar/Library/Application                                                                 │
│ Support/pyapp/ddev/10394356751862803079/11.2.0/lib/python3.12/site-packages/ddev/cli/application │
│ .py:76 in set_repo                                                                               │
│                                                                                                  │
│    73 │   │   elif repo := os.environ.get(AppEnvVars.REPO, ''):                                  │
│    74 │   │   │   self.__repo = Repository(repo, self.config.repos[repo])                        │
│    75 │   │   else:                                                                              │
│ ❱  76 │   │   │   self.__repo = Repository(self.config.repo.name, self.config.repo.path)         │
│    77 │   │                                                                                      │
│    78 │   │   self.__github = GitHubManager(                                                     │
│    79 │   │   │   self.repo, user=self.config.github.user, token=self.config.github.token, sta   │
│                                                                                                  │
│ /Users/edgar/Library/Application                                                                 │
│ Support/pyapp/ddev/10394356751862803079/11.2.0/lib/python3.12/site-packages/ddev/config/model.py │
│ :82 in repo                                                                                      │
│                                                                                                  │
│    79 │   │   │   if not isinstance(repo, str):                                                  │
│    80 │   │   │   │   self.raise_error('must be a string')                                       │
│    81 │   │   │   elif repo not in self.repos:                                                   │
│ ❱  82 │   │   │   │   self.raise_error('unknown repository')                                     │
│    83 │   │   │                                                                                  │
│    84 │   │   │   self.raw_data['repo'] = repo                                                   │
│    85 │   │   │   self._field_repo = RepoConfig({'name': repo, 'path': self.repos[repo]}, ('re   │
│                                                                                                  │
│ /Users/edgar/Library/Application                                                                 │
│ Support/pyapp/ddev/10394356751862803079/11.2.0/lib/python3.12/site-packages/ddev/config/model.py │
│ :57 in raise_error                                                                               │
│                                                                                                  │
│    54 │   │   import inspect                                                                     │
│    55 │   │                                                                                      │
│    56 │   │   field = inspect.currentframe().f_back.f_code.co_name                               │
│ ❱  57 │   │   raise ConfigurationError(message, location=' -> '.join([*self.steps, field, *ext   │
│    58                                                                                            │
│    59                                                                                            │
│    60 class RootConfig(LazilyParsedConfig):                                                      │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
ConfigurationError: Error parsing config:
repo
  unknown repository
 ~/repos/integrations-core  master ?1  ddev config set repos.core .                                                                                                                                                            ✔  15:10:07
╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
│ /Users/edgar/Library/Application                                                                 │
│ Support/pyapp/ddev/10394356751862803079/11.2.0/lib/python3.12/site-packages/ddev/cli/__init__.py │
│ :165 in main                                                                                     │
│                                                                                                  │
│   162 │   manager.hook.register_commands()                                                       │
│   163 │                                                                                          │
│   164 │   try:                                                                                   │
│ ❱ 165 │   │   return ddev(prog_name='ddev', windows_expand_args=False)                           │
│   166 │   except Exception:                                                                      │
│   167 │   │   from rich.console import Console                                                   │
│   168                                                                                            │
│                                                                                                  │
│ /Users/edgar/Library/Application                                                                 │
│ Support/pyapp/ddev/10394356751862803079/11.2.0/lib/python3.12/site-packages/click/core.py:1161   │
│ in __call__                                                                                      │
│                                                                                                  │
│ /Users/edgar/Library/Application                                                                 │
│ Support/pyapp/ddev/10394356751862803079/11.2.0/lib/python3.12/site-packages/click/core.py:1082   │
│ in main                                                                                          │
│                                                                                                  │
│ /Users/edgar/Library/Application                                                                 │
│ Support/pyapp/ddev/10394356751862803079/11.2.0/lib/python3.12/site-packages/click/core.py:1694   │
│ in invoke                                                                                        │
│                                                                                                  │
│ /Users/edgar/Library/Application                                                                 │
│ Support/pyapp/ddev/10394356751862803079/11.2.0/lib/python3.12/site-packages/click/core.py:1443   │
│ in invoke                                                                                        │
│                                                                                                  │
│ /Users/edgar/Library/Application                                                                 │
│ Support/pyapp/ddev/10394356751862803079/11.2.0/lib/python3.12/site-packages/click/core.py:788 in │
│ invoke                                                                                           │
│                                                                                                  │
│ /Users/edgar/Library/Application                                                                 │
│ Support/pyapp/ddev/10394356751862803079/11.2.0/lib/python3.12/site-packages/click/decorators.py: │
│ 33 in new_func                                                                                   │
│                                                                                                  │
│ /Users/edgar/Library/Application                                                                 │
│ Support/pyapp/ddev/10394356751862803079/11.2.0/lib/python3.12/site-packages/ddev/cli/__init__.py │
│ :133 in ddev                                                                                     │
│                                                                                                  │
│   130 │   │   │   app.display_warning(error)                                                     │
│   131 │                                                                                          │
│   132 │   # Do this last                                                                         │
│ ❱ 133 │   app.set_repo(core, extras, marketplace, agent, here)                                   │
│   134 │                                                                                          │
│   135 │   # TODO: remove this when the old CLI is gone                                           │
│   136 │   app.initialize_old_cli()                                                               │
│                                                                                                  │
│ /Users/edgar/Library/Application                                                                 │
│ Support/pyapp/ddev/10394356751862803079/11.2.0/lib/python3.12/site-packages/ddev/cli/application │
│ .py:76 in set_repo                                                                               │
│                                                                                                  │
│    73 │   │   elif repo := os.environ.get(AppEnvVars.REPO, ''):                                  │
│    74 │   │   │   self.__repo = Repository(repo, self.config.repos[repo])                        │
│    75 │   │   else:                                                                              │
│ ❱  76 │   │   │   self.__repo = Repository(self.config.repo.name, self.config.repo.path)         │
│    77 │   │                                                                                      │
│    78 │   │   self.__github = GitHubManager(                                                     │
│    79 │   │   │   self.repo, user=self.config.github.user, token=self.config.github.token, sta   │
│                                                                                                  │
│ /Users/edgar/Library/Application                                                                 │
│ Support/pyapp/ddev/10394356751862803079/11.2.0/lib/python3.12/site-packages/ddev/config/model.py │
│ :82 in repo                                                                                      │
│                                                                                                  │
│    79 │   │   │   if not isinstance(repo, str):                                                  │
│    80 │   │   │   │   self.raise_error('must be a string')                                       │
│    81 │   │   │   elif repo not in self.repos:                                                   │
│ ❱  82 │   │   │   │   self.raise_error('unknown repository')                                     │
│    83 │   │   │                                                                                  │
│    84 │   │   │   self.raw_data['repo'] = repo                                                   │
│    85 │   │   │   self._field_repo = RepoConfig({'name': repo, 'path': self.repos[repo]}, ('re   │
│                                                                                                  │
│ /Users/edgar/Library/Application                                                                 │
│ Support/pyapp/ddev/10394356751862803079/11.2.0/lib/python3.12/site-packages/ddev/config/model.py │
│ :57 in raise_error                                                                               │
│                                                                                                  │
│    54 │   │   import inspect                                                                     │
│    55 │   │                                                                                      │
│    56 │   │   field = inspect.currentframe().f_back.f_code.co_name                               │
│ ❱  57 │   │   raise ConfigurationError(message, location=' -> '.join([*self.steps, field, *ext   │
│    58                                                                                            │
│    59                                                                                            │
│    60 class RootConfig(LazilyParsedConfig):                                                      │
╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
ConfigurationError: Error parsing config:
repo
  unknown repository
  ```
  
  So:
  
  ```
  ConfigurationError: Error parsing config:
repo
  unknown repository
 ```
 
 I tried reinstalling, but even after a fresh install I'm getting the same error, so I'm guessing there's still a file somewhere that it's not being cleaned up when I do uninstall `ddev`.
 
 ```
 sudo rm -rf /usr/local/ddev/bin/ddev
 sudo rm -rf /Users/edgar/Library/Application\ Support/pyapp/ddev
 ```
 
 Could you help me please? I wanted te implement a test scenario to cover it, but I haven't been able yet.

edgrz commented:
> Hi @dkirov-dd,
> 
> Thanks for the review.
> 
> I'm using using rabbitMQ v4, which already exposes those. It was already part of v4.0.0, [see](https://github.com/rabbitmq/rabbitmq-server/blob/v4.0.0/deps/rabbitmq_prometheus/src/collectors/prometheus_rabbitmq_core_metrics_collector.erl#L224).
> 
> I updated manually the csv, but I'm facing an issue when setting up the dev environment locally following [docs](https://datadoghq.dev/integrations-core/setup/). By accident, when running the command `ddev config set repos.core /path/to/integrations-core` I set a wrong path and in order to fix I just run `ddev config edit` and removed the `repos.core` section. Since that moment, I always receive the following error:
> 
> ```
> ddev config set repos.core                                                                                                                                                              ✔  15:10:05
> ╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
> │ /Users/edgar/Library/Application                                                                 │
> │ Support/pyapp/ddev/10394356751862803079/11.2.0/lib/python3.12/site-packages/ddev/cli/__init__.py │
> │ :165 in main                                                                                     │
> │                                                                                                  │
> │   162 │   manager.hook.register_commands()                                                       │
> │   163 │                                                                                          │
> │   164 │   try:                                                                                   │
> │ ❱ 165 │   │   return ddev(prog_name='ddev', windows_expand_args=False)                           │
> │   166 │   except Exception:                                                                      │
> │   167 │   │   from rich.console import Console                                                   │
> │   168                                                                                            │
> │                                                                                                  │
> │ /Users/edgar/Library/Application                                                                 │
> │ Support/pyapp/ddev/10394356751862803079/11.2.0/lib/python3.12/site-packages/click/core.py:1161   │
> │ in __call__                                                                                      │
> │                                                                                                  │
> │ /Users/edgar/Library/Application                                                                 │
> │ Support/pyapp/ddev/10394356751862803079/11.2.0/lib/python3.12/site-packages/click/core.py:1082   │
> │ in main                                                                                          │
> │                                                                                                  │
> │ /Users/edgar/Library/Application                                                                 │
> │ Support/pyapp/ddev/10394356751862803079/11.2.0/lib/python3.12/site-packages/click/core.py:1694   │
> │ in invoke                                                                                        │
> │                                                                                                  │
> │ /Users/edgar/Library/Application                                                                 │
> │ Support/pyapp/ddev/10394356751862803079/11.2.0/lib/python3.12/site-packages/click/core.py:1443   │
> │ in invoke                                                                                        │
> │                                                                                                  │
> │ /Users/edgar/Library/Application                                                                 │
> │ Support/pyapp/ddev/10394356751862803079/11.2.0/lib/python3.12/site-packages/click/core.py:788 in │
> │ invoke                                                                                           │
> │                                                                                                  │
> │ /Users/edgar/Library/Application                                                                 │
> │ Support/pyapp/ddev/10394356751862803079/11.2.0/lib/python3.12/site-packages/click/decorators.py: │
> │ 33 in new_func                                                                                   │
> │                                                                                                  │
> │ /Users/edgar/Library/Application                                                                 │
> │ Support/pyapp/ddev/10394356751862803079/11.2.0/lib/python3.12/site-packages/ddev/cli/__init__.py │
> │ :133 in ddev                                                                                     │
> │                                                                                                  │
> │   130 │   │   │   app.display_warning(error)                                                     │
> │   131 │                                                                                          │
> │   132 │   # Do this last                                                                         │
> │ ❱ 133 │   app.set_repo(core, extras, marketplace, agent, here)                                   │
> │   134 │                                                                                          │
> │   135 │   # TODO: remove this when the old CLI is gone                                           │
> │   136 │   app.initialize_old_cli()                                                               │
> │                                                                                                  │
> │ /Users/edgar/Library/Application                                                                 │
> │ Support/pyapp/ddev/10394356751862803079/11.2.0/lib/python3.12/site-packages/ddev/cli/application │
> │ .py:76 in set_repo                                                                               │
> │                                                                                                  │
> │    73 │   │   elif repo := os.environ.get(AppEnvVars.REPO, ''):                                  │
> │    74 │   │   │   self.__repo = Repository(repo, self.config.repos[repo])                        │
> │    75 │   │   else:                                                                              │
> │ ❱  76 │   │   │   self.__repo = Repository(self.config.repo.name, self.config.repo.path)         │
> │    77 │   │                                                                                      │
> │    78 │   │   self.__github = GitHubManager(                                                     │
> │    79 │   │   │   self.repo, user=self.config.github.user, token=self.config.github.token, sta   │
> │                                                                                                  │
> │ /Users/edgar/Library/Application                                                                 │
> │ Support/pyapp/ddev/10394356751862803079/11.2.0/lib/python3.12/site-packages/ddev/config/model.py │
> │ :82 in repo                                                                                      │
> │                                                                                                  │
> │    79 │   │   │   if not isinstance(repo, str):                                                  │
> │    80 │   │   │   │   self.raise_error('must be a string')                                       │
> │    81 │   │   │   elif repo not in self.repos:                                                   │
> │ ❱  82 │   │   │   │   self.raise_error('unknown repository')                                     │
> │    83 │   │   │                                                                                  │
> │    84 │   │   │   self.raw_data['repo'] = repo                                                   │
> │    85 │   │   │   self._field_repo = RepoConfig({'name': repo, 'path': self.repos[repo]}, ('re   │
> │                                                                                                  │
> │ /Users/edgar/Library/Application                                                                 │
> │ Support/pyapp/ddev/10394356751862803079/11.2.0/lib/python3.12/site-packages/ddev/config/model.py │
> │ :57 in raise_error                                                                               │
> │                                                                                                  │
> │    54 │   │   import inspect                                                                     │
> │    55 │   │                                                                                      │
> │    56 │   │   field = inspect.currentframe().f_back.f_code.co_name                               │
> │ ❱  57 │   │   raise ConfigurationError(message, location=' -> '.join([*self.steps, field, *ext   │
> │    58                                                                                            │
> │    59                                                                                            │
> │    60 class RootConfig(LazilyParsedConfig):                                                      │
> ╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
> ConfigurationError: Error parsing config:
> repo
>   unknown repository
>  ~/repos/integrations-core  master ?1  ddev config set repos.core .                                                                                                                                                            ✔  15:10:07
> ╭─────────────────────────────── Traceback (most recent call last) ────────────────────────────────╮
> │ /Users/edgar/Library/Application                                                                 │
> │ Support/pyapp/ddev/10394356751862803079/11.2.0/lib/python3.12/site-packages/ddev/cli/__init__.py │
> │ :165 in main                                                                                     │
> │                                                                                                  │
> │   162 │   manager.hook.register_commands()                                                       │
> │   163 │                                                                                          │
> │   164 │   try:                                                                                   │
> │ ❱ 165 │   │   return ddev(prog_name='ddev', windows_expand_args=False)                           │
> │   166 │   except Exception:                                                                      │
> │   167 │   │   from rich.console import Console                                                   │
> │   168                                                                                            │
> │                                                                                                  │
> │ /Users/edgar/Library/Application                                                                 │
> │ Support/pyapp/ddev/10394356751862803079/11.2.0/lib/python3.12/site-packages/click/core.py:1161   │
> │ in __call__                                                                                      │
> │                                                                                                  │
> │ /Users/edgar/Library/Application                                                                 │
> │ Support/pyapp/ddev/10394356751862803079/11.2.0/lib/python3.12/site-packages/click/core.py:1082   │
> │ in main                                                                                          │
> │                                                                                                  │
> │ /Users/edgar/Library/Application                                                                 │
> │ Support/pyapp/ddev/10394356751862803079/11.2.0/lib/python3.12/site-packages/click/core.py:1694   │
> │ in invoke                                                                                        │
> │                                                                                                  │
> │ /Users/edgar/Library/Application                                                                 │
> │ Support/pyapp/ddev/10394356751862803079/11.2.0/lib/python3.12/site-packages/click/core.py:1443   │
> │ in invoke                                                                                        │
> │                                                                                                  │
> │ /Users/edgar/Library/Application                                                                 │
> │ Support/pyapp/ddev/10394356751862803079/11.2.0/lib/python3.12/site-packages/click/core.py:788 in │
> │ invoke                                                                                           │
> │                                                                                                  │
> │ /Users/edgar/Library/Application                                                                 │
> │ Support/pyapp/ddev/10394356751862803079/11.2.0/lib/python3.12/site-packages/click/decorators.py: │
> │ 33 in new_func                                                                                   │
> │                                                                                                  │
> │ /Users/edgar/Library/Application                                                                 │
> │ Support/pyapp/ddev/10394356751862803079/11.2.0/lib/python3.12/site-packages/ddev/cli/__init__.py │
> │ :133 in ddev                                                                                     │
> │                                                                                                  │
> │   130 │   │   │   app.display_warning(error)                                                     │
> │   131 │                                                                                          │
> │   132 │   # Do this last                                                                         │
> │ ❱ 133 │   app.set_repo(core, extras, marketplace, agent, here)                                   │
> │   134 │                                                                                          │
> │   135 │   # TODO: remove this when the old CLI is gone                                           │
> │   136 │   app.initialize_old_cli()                                                               │
> │                                                                                                  │
> │ /Users/edgar/Library/Application                                                                 │
> │ Support/pyapp/ddev/10394356751862803079/11.2.0/lib/python3.12/site-packages/ddev/cli/application │
> │ .py:76 in set_repo                                                                               │
> │                                                                                                  │
> │    73 │   │   elif repo := os.environ.get(AppEnvVars.REPO, ''):                                  │
> │    74 │   │   │   self.__repo = Repository(repo, self.config.repos[repo])                        │
> │    75 │   │   else:                                                                              │
> │ ❱  76 │   │   │   self.__repo = Repository(self.config.repo.name, self.config.repo.path)         │
> │    77 │   │                                                                                      │
> │    78 │   │   self.__github = GitHubManager(                                                     │
> │    79 │   │   │   self.repo, user=self.config.github.user, token=self.config.github.token, sta   │
> │                                                                                                  │
> │ /Users/edgar/Library/Application                                                                 │
> │ Support/pyapp/ddev/10394356751862803079/11.2.0/lib/python3.12/site-packages/ddev/config/model.py │
> │ :82 in repo                                                                                      │
> │                                                                                                  │
> │    79 │   │   │   if not isinstance(repo, str):                                                  │
> │    80 │   │   │   │   self.raise_error('must be a string')                                       │
> │    81 │   │   │   elif repo not in self.repos:                                                   │
> │ ❱  82 │   │   │   │   self.raise_error('unknown repository')                                     │
> │    83 │   │   │                                                                                  │
> │    84 │   │   │   self.raw_data['repo'] = repo                                                   │
> │    85 │   │   │   self._field_repo = RepoConfig({'name': repo, 'path': self.repos[repo]}, ('re   │
> │                                                                                                  │
> │ /Users/edgar/Library/Application                                                                 │
> │ Support/pyapp/ddev/10394356751862803079/11.2.0/lib/python3.12/site-packages/ddev/config/model.py │
> │ :57 in raise_error                                                                               │
> │                                                                                                  │
> │    54 │   │   import inspect                                                                     │
> │    55 │   │                                                                                      │
> │    56 │   │   field = inspect.currentframe().f_back.f_code.co_name                               │
> │ ❱  57 │   │   raise ConfigurationError(message, location=' -> '.join([*self.steps, field, *ext   │
> │    58                                                                                            │
> │    59                                                                                            │
> │    60 class RootConfig(LazilyParsedConfig):                                                      │
> ╰──────────────────────────────────────────────────────────────────────────────────────────────────╯
> ConfigurationError: Error parsing config:
> repo
>   unknown repository
> ```
> 
> So:
> 
> ```
> ConfigurationError: Error parsing config:
> repo
> unknown repository
> ```
> 
> I tried reinstalling, but even after a fresh install I'm getting the same error, so I'm guessing there's still a file somewhere that it's not being cleaned up when I do uninstall `ddev`.
> 
> ```
> sudo rm -rf /usr/local/ddev/bin/ddev
> sudo rm -rf /Users/edgar/Library/Application\ Support/pyapp/ddev
> ```
> 
> Could you help me please? I wanted te implement a test scenario to cover it, but I haven't been able yet.

nevermind @dkirov-dd, I managed to fix it. I had to also remove an extra file `config.tom` under `dd-checks-dev` user config folder.

I'll introduce rabbitMQ 4.0 version in the `hatch.toml` matrix and implement a new test for the new set of metrics.

dkirov-dd commented:
Nice! Let me know if you need any help from my side.

edgrz commented:
Hi @dkirov-dd,

I have introduced the v4.0 testing, as well as appended to existing `rabbitmq/tests/fixtures` the corresponding dataseries to make tests pass for v4.0.

I introduced a new test scenario that covers the new detailed endpoint with a `skipif` decorator, to not being executed when having `RABBITMQ_VERSION` is older than 4.0, as the new unaggregated endpoint is available only since v4.0.

I had to modify the `per-objects.txt` and `metrics.txt` to include the new metrics, but since those are shared across versions, there are a couple of scenarios that are not passing for 3.11-openmetrics, as the new dataseries introduced are not "assessed". 

Could you maybe help me or make the necessary fixes please?

dkirov-dd commented:
I fixed the tests. 
Thanks again @edgrz for making this PR!

@iliakur could you have a look?

edgrz commented:
thanks @dkirov-dd for fixing the tests. Looking forward for this PR to be merged.

edgrz commented:
Hi @dkirov-dd @iliakur could you give a ballpark estimation when this could be potentially released once merged? We are in the need of this metrics for autoscaling our workloads, as well as, monitor them so it would be super nice if could be included in the next release 🙏

dkirov-dd commented:
@edgrz We'll shoot for the next release (7.66)

NouemanKHAL commented on rabbitmq/tests/test_openmetrics.py:
```diff
@@ -142,13 +143,60 @@ def test_unaggregated_endpoint(endpoint, fixture_file, expected_metrics, aggrega
+def test_unaggregated_endpoint_v4(
+    endpoint, fixture_file, expected_metrics, aggregator, dd_run_check, mock_http_response
+):
+    mock_http_response(file_path=OM_RESPONSE_FIXTURES / fixture_file)
```
```suggestion
    mock_http_response(file_path=os.path.join(OM_RESPONSE_FIXTURES, fixture_file))
```
To make sure this works on windows as well, let's use `os.path.join` here (make sure to add the necessary imports)

NouemanKHAL commented on rabbitmq/tests/test_openmetrics.py:
```diff
@@ -142,13 +143,60 @@ def test_unaggregated_endpoint(endpoint, fixture_file, expected_metrics, aggrega
+        pytest.param(
+            'detailed?family=queue_delivery_metrics',
+            "detailed-queue_delivery_metrics.txt",
+            {
+                'rabbitmq.queue.get.ack.count',
+                'rabbitmq.queue.get.count',
+                'rabbitmq.queue.messages.delivered.ack.count',
+                'rabbitmq.queue.messages.delivered.count',
+                'rabbitmq.queue.messages.redelivered.count',
+                'rabbitmq.queue.messages.acked.count',
+                'rabbitmq.queue.get.empty.count',
+            },
```
```suggestion
	RABBITMQ_4_0_ADDED,
```
I think this would be better and more readable.
The suggestion requires adding the necessary import `from .metrics` with `RABBITMQ_4_0_ADDED`

NouemanKHAL commented on rabbitmq/changelog.d/19963.added:
```diff
@@ -0,0 +1 @@
+RabbitMQ: Adding the `queue_delivery_metrics` OpenMetrics endpoint
```
```suggestion
Add support for queue_delivery_metrics group
```

dkirov-dd commented:
Hello @edgrz, this change will be included in the `Agent 7.66` release.
Thanks again for your contribution!

edgrz commented:
thanks @dkirov-dd for the review and help!
